### PR TITLE
tests suite reporter: Collect guest VM cloud-init logs

### PR DESCRIPTION
**Pull request by ormergi**
Signed-off-by: Or Mergi <ormergi@redhat.com>

**What this PR does / why we need it**:
With this PR test suite reporter will collect cloud-init logs from guest VM's from failing tests
and save them at `artifacts/cloud-init/vmis`.

Having those logs will help debugging failing tests due to cloud-init configuration failures.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
This PR changes will help with working on https://github.com/kubevirt/kubevirt/issues/6776

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -1115,6 +1174,37 @@ func (r *KubernetesReporter) executePriviledgedVirtLauncherCommands(virtCli kube
 	}
 }
 
+func (r *KubernetesReporter) executeCloudInitCommands(vmi v12.VirtualMachineInstance, path string, vmiType string) {
```
Comment (maya-r) [inline tests/reporter/kubernetes.go:1177]: this file could use better names for functions, but I guess it is consistent with existing usage and it is best handled in a separate PR.

Comment (ormergi) [reply tests/reporter/kubernetes.go:1177]: Yep, I am planning to post small refactoring PR for this file soon, we can do it then :slightly_smiling_face:

**Comment by ormergi** [comment]
/hold
I will remove it when finish verifying that we get the files we need

**Comment by ormergi** [comment]
/unhold
Verified by failing one of the tests on local environment, we get the cloud-init logs as expected

**Comment by rmohr** [comment]
/approve

Looks useful :+1:

**Comment by ormergi** [comment]
Thank you :slightly_smiling_face:

**Pull request merged**